src/api.js

```diff
@@ -74,8 +76,74 @@ module.exports = {
 				self.checkVariables();
 			})
 			.on('error', function(error) {
+				this.updateStatus(InstanceStatus.ConnectionFailure, error.message)
 				self.log('error', 'Error Sending Command ' + error.toString());
 			});
 		}
+	},
+	
+	sendPostRequest(path, formData) {
+		if (this.config.host !== '' && this.config.host !== undefined) {
```

Comment:
here in `sendPostRequest` and `processResponse`, consider using `let self = this` and use `self` instead of `this`, for consistency/readability across the module. This will work, but it is not consistent.